### PR TITLE
Feature/pytorch load tfrecord

### PR DESCRIPTION
**Pull request by wenshuoliu**
This PR adds option to read .tfrecord data and train in pytorch. 

This implementation uses `tfrecord.reader.tfrecord_loader` to read file. In this way, it reads multiple files in exactly the same way as the other `MultiFileLoader`. 

I had another implementation that uses `tfrecord.reader.multi_tfrecord_loader`, where reading multiple files relies on `tfrecord` module's implementation. In this way, it's hard to turn off `shuffle`. But IMO `shuffle == False` is only useful when testing response selection performance. I can do it this way if you want.

The real problem here is that I don't observe any speeding up in training.

### CODE REVIEW

File: api-examples/transformer_utils.py [hunk 1]
```diff
@@ -13,7 +13,7 @@
 from collections import Counter
 import glob
 import json
-
+import tfrecord
```
Comment (dpressel) [inline api-examples/transformer_utils.py:16]: this creates a dependency in the entire file, do we really want that?

File: api-examples/pretrain_tlm_pytorch.py [hunk 2]
```diff
@@ -102,7 +102,12 @@ def train():
         args.device, updated_local_rank = init_distributed(args.local_rank)
         args.local_rank = updated_local_rank
 
-    reader_type = "lang" if not args.preprocessed else "preprocessed"
+    if args.pattern == '*.tfrecord':
```
Comment (dpressel) [inline api-examples/pretrain_tlm_pytorch.py:105]: In the TF version, I used a file_type, I think that makes more sense than the way we are doing it here now

Comment (dpressel) [reply api-examples/pretrain_tlm_pytorch.py:105]: to get the pattern from that is easy
```
    pattern = os.path.join(directory, f'*.{file_type}')
```

File: api-examples/transformer_utils.py [hunk 3]
```diff
@@ -495,6 +498,40 @@ def process_line(self, line):
...
+            for file_idx in read_file_order:
+                file = files[file_idx]
+                idx_file = '.'.join(file.split('.')[:-1]) + '.index'
+                # shard = (worker_id, num_workers), but we already assigned this file to one certain worker,
+                # so shard = (0, 1)
+                itr = tfrecord.reader.tfrecord_loader(file, idx_file, shard=(0, 1))
+                if self.shuffle:
+                    np.random.seed(node_worker_id)
+                    # not sure about the optimal choice of shuffle_queue_size here:
+                    itr = tfrecord.iterator_utils.shuffle_iterator(itr, queue_size=128)
```
Comment (dpressel) [inline api-examples/transformer_utils.py:527]: seems a bit too small

File: api-examples/transformer_utils.py [hunk 4]
```diff
@@ -517,6 +554,9 @@ def load(self, directory, vocabs, distribute=True, shuffle=True):
         elif reader_type == "lang":
             print("Using files as an LM")
             return SequencePredictionFileLoader(directory, self.pattern, vocabs, self.vectorizer, self.nctx, distribute=distribute, shuffle=shuffle)
+        elif reader_type == 'tfrecord':
+            print("Reading data in .tfrecord fomat using the tfrecord module")
```
Comment (dpressel) [inline api-examples/transformer_utils.py:558]: spelling
